### PR TITLE
Add support for good and neutral tests in YAML-based testing

### PR DESCRIPTION
**Pull request by srikanth-kumar**
## ✨ Summary

This PR updates the YAML-based testing framework in GeoIPS with: 

**New Features:**

-  Add Pydantic validation for the YAML-based test cases
    - For further reference, see issue 
      - #1069 
- Add `description` field and associated logic to YAML-based test case inputs
  - further reference, see issue 
    - #1065 
- Neutral test case:  A new test category to test validate deprecation warnings and log-level warnings. 
- Good test case:   A new test category for multiple good test cases, including edge cases.
  - The term 'Good' was previously used to represent a test that validated the default valid plugin definition. 

**Enhancements:**

- Refactors file path construction to use  `pathlib` instead of f-strings method to align with the best practices. 
  - Further reference, see issue 
    - #1072 
- Base test case:  The test case previously labeled 'Good' has been renamed to 'Base' to indicate that it tests the default valid plugin definition as a single-use case (one known-good configuration). 

YAML-based testing-Version-02

---

## ✅ Checklist for Reviewer Reference (filled out by PR Author)

- [x] **Tests**: All tests and CI pass (e.g., full, base, extra, etc.)
    - [x] **Full test**: Yes, full test *is passing*.
- [x] **Integration Tests**: Integration tests added/updated for new/modified functionality *OR* no new functionality
- [x] **Other Repos**: I have updated other repositories to handle this change *OR* my package doesn't impact other plugin packages.

For more details, please see our [review template](https://github.com/NRLMMD-GEOIPS/.github/blob/main/.github/review-template.md) 💌

---

## 🔗 Related Issues

- **Fixes:** #1065 
- **Fixes:** #1069
- **Fixes:** #1072
- **Fixes:** #1075 

  

---

## 🧪 Testing Instructions

Run pytest from the `~/geoips/geoips/tests/unit_tests/pydantic` directory.

---

## Test Output

image

### CODE REVIEW

File: docs/source/releases/latest/1001-add-support-for-neutral-tests-in-yaml-based-testing.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    - 'tests/unit_tests/pydantic/feature_annotators/test_cases_bad.yaml'
+    - 'tests/unit_tests/pydantic/feature_annotators/test_cases_neutral.yaml'
+    - 'tests/unit_tests/pydantic/sectors/test_cases_bad.yaml'
+    - 'tests/unit_tests/pydantic/sectors/test_cases_neutral.yaml'
+    modified:
+    - 'tests/unit_tests/pydantic/feature_annotators/test_feature_annotators.py'
+    - 'tests/unit_tests/pydantic/sectors/test_sectors.py'
+    - 'tests/unit_tests/pydantic/utils.py'
+  related-issue:
+    number: 1065, 1069, 1072
```
Comment (biosafetylvl5) [inline docs/source/releases/latest/1001-add-support-for-neutral-tests-in-yaml-based-testing.yaml:21]: This wont work until the new brassy version comes out. I suggest breaking this up into atomic chunks for now

File: tests/unit_tests/pydantic/utils.py [hunk 2]
```diff
@@ -64,34 +99,43 @@ def load_test_cases(interface_name):
...
+    fname = f"test_cases_{test_type}.yaml"
+    base_dir = Path(__file__).parent
+    fpath = base_dir / interface_name / fname
+
+    if not fpath.exists():
         raise FileNotFoundError(
-            f"Error: No test cases file could be found. Expected {fpath} but it did not"
-            " exist. Please create this file and rerun your tests."
+            f"Error: No test cases file could be found. Expected {fpath} but it "
+            "did not exist. Please create this file and rerun your tests."
```
Comment (biosafetylvl5) [inline tests/unit_tests/pydantic/utils.py:120]: grammar: does not

**Review by biosafetylvl5 — CHANGES_REQUESTED** [review]
Small changes requested - happy to re-review once the dev-staging tests are passing.

File: tests/unit_tests/pydantic/utils.py (hunk 2 above)
Comment (srikanth-kumar) [reply tests/unit_tests/pydantic/utils.py]: This has been fixed as suggested and is now ready to be resolved.

File: docs/source/releases/latest/1001-add-support-for-neutral-tests-in-yaml-based-testing.yaml (hunk 1 above)
Comment (srikanth-kumar) [reply docs/source/releases/latest/1001-add-support-for-neutral-tests-in-yaml-based-testing.yaml]: I've created a primary issue and added the three issues above as sub-issues under it. The release notes have been updated accordingly.

**Review by jsolbrig — DISMISSED** [review]
I'm approving this but, since this test infrastructure seems to be firming up, I would like to see documentation for how this test infrastructure works. Can you please create an issue for the documentation before merging this?

**Review by jsolbrig — APPROVED** [review]

**Review by mindyls — APPROVED** [review]
